File: Grind169/Week2/53-Maximum-Subarray/solution.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution{
public:
    int maxSubArray(vector<int>&nums){
        for (int i=1;i<nums.size();i++){
            if (nums[i-1]>0){
                nums[i]+=nums[i-1];
            }
        }
        int max=*max_element(nums.begin(),nums.end());
        return max;
    }
};

File: Grind169/Week2/53-Maximum-Subarray/solution.py
from typing import List


# Kadane's Algorithm
# https://shubo.io/maximum-subarray-problem-kadane-algorithm/
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        for i in range(1, len(nums)):
            if nums[i - 1] > 0:
                nums[i] += nums[i - 1]
        return max(nums)
